assignment07/script.js

```diff
@@ -0,0 +1,110 @@
+// 1.
+//  How will you create a new copy of the object below while updating the value of address.details[0] to “5“?
+// {
+//     name:”Harry Potter”,
+//     age: 12,
+//         address: {
+//         details: [“4”, “Privet Drive”],
+//             area:”Little Whinging”,
+//         city: “Surrey”,
+//         state: “England”
+//     }
+// }
+
+    // -> first approach using nested spread operator
+
+        var obj1 = {
+            name:"Harry Potter",
+            age: 12,
+            address: {
+                details: ["4", "Privet Drive"],
+                area:"Little Whinging",
+                city: "Surrey",
+                state: "England"    
+            }  
+        }   
+        
+        var obj2  = {...obj1,
+            address:{...obj1.address, 
+                    details:[...obj1.address.details]
+                }
+        }
+
+        console.log(obj1)
+        console.log(obj2)
+        
+        if(obj1==obj2){
+            console.log("obj1 is same as obj2")
+        }
+        else {
+            // still giving this, why?
```

Comment:
You can't compare objects using ==  operator as objects/arrays are passed & handled in JS using their reference address. You can use JSON.stringify(obj1) === JSON.stringify(obj2) as the easiest method or else use deepCompare function of lodash library

Reply:
Understood